apple/REAModule.mm

```diff
@@ -158,47 +159,35 @@ - (void)handleJavaScriptDidLoadNotification:(NSNotification *)notification
 
 /*
  * Taken from RCTNativeAnimatedTurboModule:
- * In bridgeless mode, `setBridge` is never called during initialization. Instead this selector is invoked via
- * BridgelessTurboModuleSetup.
+ * This selector is invoked via BridgelessTurboModuleSetup.
  */
 - (void)setSurfacePresenter:(id<RCTSurfacePresenterStub>)surfacePresenter
 {
   _surfacePresenter = surfacePresenter;
+  _isBridgeless = true;
 }
 
 - (void)setBridge:(RCTBridge *)bridge
 {
-  // This method isn't called on Bridgeless mode.
   [super setBridge:bridge];
-
-  [bridge.uiManager.observerCoordinator addObserver:self];
-
   // only within the first loading `self.bridge.surfacePresenter` exists
   // during the reload `self.bridge.surfacePresenter` is null
-  _surfacePresenter = self.bridge.surfacePresenter;
+  if (self.bridge.surfacePresenter) {
+    _surfacePresenter = self.bridge.surfacePresenter;
+  }
 
 #ifndef NDEBUG
   [self setReaSurfacePresenter];
 #endif // NDEBUG
 
-  [self setNodesManager:self.bridge];
-}
+  _nodesManager = [[REANodesManager alloc] initWithModule:self bridge:bridge surfacePresenter:_surfacePresenter];
 
-- (void)initialize
-{
   [[NSNotificationCenter defaultCenter] addObserver:self
                                            selector:@selector(handleJavaScriptDidLoadNotification:)
                                                name:RCTJavaScriptDidLoadNotification
                                              object:nil];
 
   [[self.moduleRegistry moduleForName:"EventDispatcher"] addDispatchObserver:self];
-
-//  [bridge.uiManager.observerCoordinator addObserver:self]; // TODO: Check if it's needed on new arch.
-#ifndef NDEBUG
-  [self setReaSurfacePresenter];
-#endif // NDEBUG
```

Comment:
I wonder why is this code DEBUG-only? It was like that before, just curious to know if that's right.

Reply:
It shouldn't be 🫣 I fixed it in https://github.com/software-mansion/react-native-reanimated/pull/5901/commits/3e54ad02b263b1d087582efe9e4f609efaa8b30f. Also. We still need `REAInitializerRCTFabricSurface` class for the Fabric with bridge :/